crates/bevy_asset/src/assets.rs

```diff
@@ -192,6 +198,16 @@ impl<T: Asset> Assets<T> {
     pub fn is_empty(&self) -> bool {
         self.assets.is_empty()
     }
+
+    /// Returns a reference to the inner assets `HashMap`.
+    pub fn assets(&self) -> &HashMap<HandleId, T> {
+        &self.assets
+    }
+
+    /// Returns a mutable reference to the inner assets `HashMap`.
+    pub fn assets_mut(&mut self) -> &mut HashMap<HandleId, T> {
+        &mut self.assets
+    }
```

Comment:
We shouldn't add these methods. HashMap is an internal implementation detail that should be hidden from users. Directly exposing access allows people to interact with the storage without going through the other access methods:

Ex: we produce events when assets are mutated, which we can't do if users directly access the HashMap.

Exposing these also prevents us from building other "encapsulated" guarantees in the future.